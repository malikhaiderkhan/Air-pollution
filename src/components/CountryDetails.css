.country-details {
  color: white;
  margin-bottom: 40px;
}

.country-detail {
  background-color: #fc2d7c;
  margin: 0;
  height: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.country-detail:nth-child(even) {
  background-color: #f5186d;
}

.country-flag {
  margin: 20px 0 40px 0;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.8);
}
