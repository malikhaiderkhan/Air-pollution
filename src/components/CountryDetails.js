import React from 'react';
import { useSelector } from 'react-redux';

function CountryDetails() {
  const selectedCountry = useSelector((state) => state.countries.selectedCountry);

  if (!selectedCountry) {
    return <p>Select a country to see details.</p>;
  }

  return (
    <div>
      <h2>{selectedCountry.name.common}</h2>
      <p>
        Population:
        {selectedCountry.population}
      </p>
      <p>
        Region:
        {selectedCountry.region}
      </p>
      <p>
        Subregion:
        {selectedCountry.subregion}
      </p>
      {/* Add more details as needed */}
    </div>
  );
}

export default CountryDetails;
