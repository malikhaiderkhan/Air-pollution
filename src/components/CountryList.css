.country-grid {
    display: grid;
    grid-template-columns: repeat(4, 1fr); /* Create four columns */
    gap: 0px; /* Add gap between grid items */
    margin: 0 20px 0 20px;
  }
  
  .country-box {
    border: 1px solid #fc2d72;
    padding: 1rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    background-color: #fc2d7c;
  }

  .country-box:nth-child(odd) {
    background-color: #f5186d; /* Light pink for odd-indexed boxes */
  }
  
  .country-button {
    text-decoration: none;
    color: #333;
  }
  
  .country-button img {
    max-width: 100px;
    height: auto;
    margin-bottom: 0.5rem;
  }
