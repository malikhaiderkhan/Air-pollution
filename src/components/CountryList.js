import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchCountryData } from '../redux/countries/countriesSlice';
import './CountryList.css';

function CountryList() {
  const dispatch = useDispatch();
  const countries = useSelector((state) => state.countries.countries);
  const loading = useSelector((state) => state.countries.loading);

  useEffect(() => {
    dispatch(fetchCountryData());
  }, [dispatch]);

  return (
    <div>
      <h2>Country List</h2>
      {loading === 'pending' && <p>Loading...</p>}
      <div className="country-grid">
        {countries.map((country) => (
          <div key={country.cca3} className="country-box">
            <Link to={`/country/${country.cca3}`} className="country-button">
              <img src={country.flags[1]} alt={country.name.common} />
              <h3>{country.name.common}</h3>
              <p>
                Population:
                {country.population}
              </p>
            </Link>
          </div>
        ))}
      </div>
    </div>
  );
}

export default CountryList;
