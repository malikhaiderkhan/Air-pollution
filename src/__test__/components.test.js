import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import {
  MemoryRouter,
} from 'react-router-dom';
import CountryList from '../components/CountryList';
import CountryDetails from '../components/CountryDetails';

const mockStore = configureStore([thunk]);

describe('Component testing', () => {
  test('Country list should render correctly', () => {
    const mockState = {
      countries: {
        countries: [
          {
            name: {
              common: 'Ecuador',
              official: 'Republic of Ecuador',
              nativeName: {
                spa: {
                  official: 'República del Ecuador',
                  common: 'Ecuador',
                },
              },
            },
            tld: [
              '.ec',
            ],
            cca2: 'EC',
            ccn3: '218',
            cca3: 'ECU',
            cioc: 'ECU',
            independent: true,
            status: 'officially-assigned',
            unMember: true,
            currencies: {
              USD: {
                name: 'United States dollar',
                symbol: '$',
              },
            },
            idd: {
              root: '+5',
              suffixes: [
                '93',
              ],
            },
            capital: [
              'Quito',
            ],
            altSpellings: [
              'EC',
              'Republic of Ecuador',
              'República del Ecuador',
            ],
            region: 'Americas',
            subregion: 'South America',
            languages: {
              spa: 'Spanish',
            },
            translations: {
              ara: {
                official: 'جمهورية الإكوادور',
                common: 'الإكوادور',
              },
              bre: {
                official: 'Republik Ecuador',
                common: 'Ecuador',
              },
              ces: {
                official: 'Ekvádorská republika',
                common: 'Ekvádor',
              },
              cym: {
                official: 'Gweriniaeth Ecwador',
                common: 'Ecwador',
              },
              deu: {
                official: 'Republik Ecuador',
                common: 'Ecuador',
              },
              est: {
                official: 'Ecuadori Vabariik',
                common: 'Ecuador',
              },
              fin: {
                official: 'Ecuadorin tasavalta',
                common: 'Ecuador',
              },
              fra: {
                official: "République de l'Équateur",
                common: 'Équateur',
              },
              hrv: {
                official: 'Republika Ekvador',
                common: 'Ekvador',
              },
              hun: {
                official: 'Ecuadori Köztársaság',
                common: 'Ecuador',
              },
              ita: {
                official: "Repubblica dell'Ecuador",
                common: 'Ecuador',
              },
              jpn: {
                official: 'エクアドル共和国',
                common: 'エクアドル',
              },
              kor: {
                official: '에콰도르 공화국',
                common: '에콰도르',
              },
              nld: {
                official: 'Republiek Ecuador',
                common: 'Ecuador',
              },
              per: {
                official: 'جمهوری اکوادور',
                common: 'اکوادور',
              },
              pol: {
                official: 'Ekwador',
                common: 'Ekwador',
              },
              por: {
                official: 'República do Equador',
                common: 'Equador',
              },
              rus: {
                official: 'Республика Эквадор',
                common: 'Эквадор',
              },
              slk: {
                official: 'Ekvádorská republika',
                common: 'Ekvádor',
              },
              spa: {
                official: 'República del Ecuador',
                common: 'Ecuador',
              },
              swe: {
                official: 'Republiken Ecuador',
                common: 'Ecuador',
              },
              urd: {
                official: 'جمہوریہ ایکوڈور',
                common: 'ایکواڈور',
              },
              zho: {
                official: '厄瓜多尔共和国',
                common: '厄瓜多尔',
              },
            },
            latlng: [
              -2,
              -77.5,
            ],
            landlocked: false,
            borders: [
              'COL',
              'PER',
            ],
            area: 276841,
            demonyms: {
              eng: {
                f: 'Ecuadorean',
                m: 'Ecuadorean',
              },
              fra: {
                f: 'Équatorienne',
                m: 'Équatorien',
              },
            },
            flag: '🇪🇨',
            maps: {
              googleMaps: 'https://goo.gl/maps/TbX8hUW4gcbRPZiK7',
              openStreetMaps: 'https://www.openstreetmap.org/relation/108089',
            },
            population: 17643060,
            gini: {
              2019: 45.7,
            },
            fifa: 'ECU',
            car: {
              signs: [
                'EC',
              ],
              side: 'right',
            },
            timezones: [
              'UTC-06:00',
              'UTC-05:00',
            ],
            continents: [
              'South America',
            ],
            flags: [
              'https://flagcdn.com/ec.svg',
              'https://flagcdn.com/w320/ec.png',
            ],
          },
          {
            name: {
              common: 'France',
              official: 'French Republic',
              nativeName: {
                fra: {
                  official: 'République française',
                  common: 'France',
                },
              },
            },
            tld: [
              '.fr',
            ],
            cca2: 'FR',
            ccn3: '250',
            cca3: 'FRA',
            cioc: 'FRA',
            independent: true,
            status: 'officially-assigned',
            unMember: true,
            currencies: {
              EUR: {
                name: 'Euro',
                symbol: '€',
              },
            },
            idd: {
              root: '+3',
              suffixes: [
                '3',
              ],
            },
            capital: [
              'Paris',
            ],
            altSpellings: [
              'FR',
              'French Republic',
              'République française',
            ],
            region: 'Europe',
            subregion: 'Western Europe',
            languages: {
              fra: 'French',
            },
            translations: {
              ara: {
                official: 'الجمهورية الفرنسية',
                common: 'فرنسا',
              },
              bre: {
                official: 'Republik Frañs',
                common: 'Frañs',
              },
              ces: {
                official: 'Francouzská republika',
                common: 'Francie',
              },
              cym: {
                official: 'French Republic',
                common: 'France',
              },
              deu: {
                official: 'Französische Republik',
                common: 'Frankreich',
              },
              est: {
                official: 'Prantsuse Vabariik',
                common: 'Prantsusmaa',
              },
              fin: {
                official: 'Ranskan tasavalta',
                common: 'Ranska',
              },
              fra: {
                official: 'République française',
                common: 'France',
              },
              hrv: {
                official: 'Francuska Republika',
                common: 'Francuska',
              },
              hun: {
                official: 'Francia Köztársaság',
                common: 'Franciaország',
              },
              ita: {
                official: 'Repubblica francese',
                common: 'Francia',
              },
              jpn: {
                official: 'フランス共和国',
                common: 'フランス',
              },
              kor: {
                official: '프랑스 공화국',
                common: '프랑스',
              },
              nld: {
                official: 'Franse Republiek',
                common: 'Frankrijk',
              },
              per: {
                official: 'جمهوری فرانسه',
                common: 'فرانسه',
              },
              pol: {
                official: 'Republika Francuska',
                common: 'Francja',
              },
              por: {
                official: 'República Francesa',
                common: 'França',
              },
              rus: {
                official: 'Французская Республика',
                common: 'Франция',
              },
              slk: {
                official: 'Francúzska republika',
                common: 'Francúzsko',
              },
              spa: {
                official: 'República francés',
                common: 'Francia',
              },
              swe: {
                official: 'Republiken Frankrike',
                common: 'Frankrike',
              },
              urd: {
                official: 'جمہوریہ فرانس',
                common: 'فرانس',
              },
              zho: {
                official: '法兰西共和国',
                common: '法国',
              },
            },
            latlng: [
              46,
              2,
            ],
            landlocked: false,
            borders: [
              'AND',
              'BEL',
              'DEU',
              'ITA',
              'LUX',
              'MCO',
              'ESP',
              'CHE',
            ],
            area: 551695,
            demonyms: {
              eng: {
                f: 'French',
                m: 'French',
              },
              fra: {
                f: 'Française',
                m: 'Français',
              },
            },
            flag: '🇫🇷',
            maps: {
              googleMaps: 'https://goo.gl/maps/g7QxxSFsWyTPKuzd7',
              openStreetMaps: 'https://www.openstreetmap.org/relation/1403916',
            },
            population: 67391582,
            gini: {
              2018: 32.4,
            },
            fifa: 'FRA',
            car: {
              signs: [
                'F',
              ],
              side: 'right',
            },
            timezones: [
              'UTC-10:00',
              'UTC-09:30',
              'UTC-09:00',
              'UTC-08:00',
              'UTC-04:00',
              'UTC-03:00',
              'UTC+01:00',
              'UTC+02:00',
              'UTC+03:00',
              'UTC+04:00',
              'UTC+05:00',
              'UTC+10:00',
              'UTC+11:00',
              'UTC+12:00',
            ],
            continents: [
              'Europe',
            ],
            flags: [
              'https://flagcdn.com/fr.svg',
              'https://flagcdn.com/w320/fr.png',
            ],
          },
        ],
        selectedCountry: null,
        loading: 'idle',
        error: null,
      },
    };

    const store = mockStore(mockState);

    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <CountryList />
        </MemoryRouter>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Country details should render correctly', () => {
    const mockState = {
      countries: {
        countries: [
          {
            name: {
              common: 'Ecuador',
              official: 'Republic of Ecuador',
              nativeName: {
                spa: {
                  official: 'República del Ecuador',
                  common: 'Ecuador',
                },
              },
            },
            tld: [
              '.ec',
            ],
            cca2: 'EC',
            ccn3: '218',
            cca3: 'ECU',
            cioc: 'ECU',
            independent: true,
            status: 'officially-assigned',
            unMember: true,
            currencies: {
              USD: {
                name: 'United States dollar',
                symbol: '$',
              },
            },
            idd: {
              root: '+5',
              suffixes: [
                '93',
              ],
            },
            capital: [
              'Quito',
            ],
            altSpellings: [
              'EC',
              'Republic of Ecuador',
              'República del Ecuador',
            ],
            region: 'Americas',
            subregion: 'South America',
            languages: {
              spa: 'Spanish',
            },
            translations: {
              ara: {
                official: 'جمهورية الإكوادور',
                common: 'الإكوادور',
              },
              bre: {
                official: 'Republik Ecuador',
                common: 'Ecuador',
              },
              ces: {
                official: 'Ekvádorská republika',
                common: 'Ekvádor',
              },
              cym: {
                official: 'Gweriniaeth Ecwador',
                common: 'Ecwador',
              },
              deu: {
                official: 'Republik Ecuador',
                common: 'Ecuador',
              },
              est: {
                official: 'Ecuadori Vabariik',
                common: 'Ecuador',
              },
              fin: {
                official: 'Ecuadorin tasavalta',
                common: 'Ecuador',
              },
              fra: {
                official: "République de l'Équateur",
                common: 'Équateur',
              },
              hrv: {
                official: 'Republika Ekvador',
                common: 'Ekvador',
              },
              hun: {
                official: 'Ecuadori Köztársaság',
                common: 'Ecuador',
              },
              ita: {
                official: "Repubblica dell'Ecuador",
                common: 'Ecuador',
              },
              jpn: {
                official: 'エクアドル共和国',
                common: 'エクアドル',
              },
              kor: {
                official: '에콰도르 공화국',
                common: '에콰도르',
              },
              nld: {
                official: 'Republiek Ecuador',
                common: 'Ecuador',
              },
              per: {
                official: 'جمهوری اکوادور',
                common: 'اکوادور',
              },
              pol: {
                official: 'Ekwador',
                common: 'Ekwador',
              },
              por: {
                official: 'República do Equador',
                common: 'Equador',
              },
              rus: {
                official: 'Республика Эквадор',
                common: 'Эквадор',
              },
              slk: {
                official: 'Ekvádorská republika',
                common: 'Ekvádor',
              },
              spa: {
                official: 'República del Ecuador',
                common: 'Ecuador',
              },
              swe: {
                official: 'Republiken Ecuador',
                common: 'Ecuador',
              },
              urd: {
                official: 'جمہوریہ ایکوڈور',
                common: 'ایکواڈور',
              },
              zho: {
                official: '厄瓜多尔共和国',
                common: '厄瓜多尔',
              },
            },
            latlng: [
              -2,
              -77.5,
            ],
            landlocked: false,
            borders: [
              'COL',
              'PER',
            ],
            area: 276841,
            demonyms: {
              eng: {
                f: 'Ecuadorean',
                m: 'Ecuadorean',
              },
              fra: {
                f: 'Équatorienne',
                m: 'Équatorien',
              },
            },
            flag: '🇪🇨',
            maps: {
              googleMaps: 'https://goo.gl/maps/TbX8hUW4gcbRPZiK7',
              openStreetMaps: 'https://www.openstreetmap.org/relation/108089',
            },
            population: 17643060,
            gini: {
              2019: 45.7,
            },
            fifa: 'ECU',
            car: {
              signs: [
                'EC',
              ],
              side: 'right',
            },
            timezones: [
              'UTC-06:00',
              'UTC-05:00',
            ],
            continents: [
              'South America',
            ],
            flags: [
              'https://flagcdn.com/ec.svg',
              'https://flagcdn.com/w320/ec.png',
            ],
          },
          {
            name: {
              common: 'France',
              official: 'French Republic',
              nativeName: {
                fra: {
                  official: 'République française',
                  common: 'France',
                },
              },
            },
            tld: [
              '.fr',
            ],
            cca2: 'FR',
            ccn3: '250',
            cca3: 'FRA',
            cioc: 'FRA',
            independent: true,
            status: 'officially-assigned',
            unMember: true,
            currencies: {
              EUR: {
                name: 'Euro',
                symbol: '€',
              },
            },
            idd: {
              root: '+3',
              suffixes: [
                '3',
              ],
            },
            capital: [
              'Paris',
            ],
            altSpellings: [
              'FR',
              'French Republic',
              'République française',
            ],
            region: 'Europe',
            subregion: 'Western Europe',
            languages: {
              fra: 'French',
            },
            translations: {
              ara: {
                official: 'الجمهورية الفرنسية',
                common: 'فرنسا',
              },
              bre: {
                official: 'Republik Frañs',
                common: 'Frañs',
              },
              ces: {
                official: 'Francouzská republika',
                common: 'Francie',
              },
              cym: {
                official: 'French Republic',
                common: 'France',
              },
              deu: {
                official: 'Französische Republik',
                common: 'Frankreich',
              },
              est: {
                official: 'Prantsuse Vabariik',
                common: 'Prantsusmaa',
              },
              fin: {
                official: 'Ranskan tasavalta',
                common: 'Ranska',
              },
              fra: {
                official: 'République française',
                common: 'France',
              },
              hrv: {
                official: 'Francuska Republika',
                common: 'Francuska',
              },
              hun: {
                official: 'Francia Köztársaság',
                common: 'Franciaország',
              },
              ita: {
                official: 'Repubblica francese',
                common: 'Francia',
              },
              jpn: {
                official: 'フランス共和国',
                common: 'フランス',
              },
              kor: {
                official: '프랑스 공화국',
                common: '프랑스',
              },
              nld: {
                official: 'Franse Republiek',
                common: 'Frankrijk',
              },
              per: {
                official: 'جمهوری فرانسه',
                common: 'فرانسه',
              },
              pol: {
                official: 'Republika Francuska',
                common: 'Francja',
              },
              por: {
                official: 'República Francesa',
                common: 'França',
              },
              rus: {
                official: 'Французская Республика',
                common: 'Франция',
              },
              slk: {
                official: 'Francúzska republika',
                common: 'Francúzsko',
              },
              spa: {
                official: 'República francés',
                common: 'Francia',
              },
              swe: {
                official: 'Republiken Frankrike',
                common: 'Frankrike',
              },
              urd: {
                official: 'جمہوریہ فرانس',
                common: 'فرانس',
              },
              zho: {
                official: '法兰西共和国',
                common: '法国',
              },
            },
            latlng: [
              46,
              2,
            ],
            landlocked: false,
            borders: [
              'AND',
              'BEL',
              'DEU',
              'ITA',
              'LUX',
              'MCO',
              'ESP',
              'CHE',
            ],
            area: 551695,
            demonyms: {
              eng: {
                f: 'French',
                m: 'French',
              },
              fra: {
                f: 'Française',
                m: 'Français',
              },
            },
            flag: '🇫🇷',
            maps: {
              googleMaps: 'https://goo.gl/maps/g7QxxSFsWyTPKuzd7',
              openStreetMaps: 'https://www.openstreetmap.org/relation/1403916',
            },
            population: 67391582,
            gini: {
              2018: 32.4,
            },
            fifa: 'FRA',
            car: {
              signs: [
                'F',
              ],
              side: 'right',
            },
            timezones: [
              'UTC-10:00',
              'UTC-09:30',
              'UTC-09:00',
              'UTC-08:00',
              'UTC-04:00',
              'UTC-03:00',
              'UTC+01:00',
              'UTC+02:00',
              'UTC+03:00',
              'UTC+04:00',
              'UTC+05:00',
              'UTC+10:00',
              'UTC+11:00',
              'UTC+12:00',
            ],
            continents: [
              'Europe',
            ],
            flags: [
              'https://flagcdn.com/fr.svg',
              'https://flagcdn.com/w320/fr.png',
            ],
          },
        ],
        selectedCountry: {
          name: {
            common: 'Ecuador',
            official: 'Republic of Ecuador',
            nativeName: {
              spa: {
                official: 'República del Ecuador',
                common: 'Ecuador',
              },
            },
          },
          tld: [
            '.ec',
          ],
          cca2: 'EC',
          ccn3: '218',
          cca3: 'ECU',
          cioc: 'ECU',
          independent: true,
          status: 'officially-assigned',
          unMember: true,
          currencies: {
            USD: {
              name: 'United States dollar',
              symbol: '$',
            },
          },
          idd: {
            root: '+5',
            suffixes: [
              '93',
            ],
          },
          capital: [
            'Quito',
          ],
          altSpellings: [
            'EC',
            'Republic of Ecuador',
            'República del Ecuador',
          ],
          region: 'Americas',
          subregion: 'South America',
          languages: {
            spa: 'Spanish',
          },
          translations: {
            ara: {
              official: 'جمهورية الإكوادور',
              common: 'الإكوادور',
            },
            bre: {
              official: 'Republik Ecuador',
              common: 'Ecuador',
            },
            ces: {
              official: 'Ekvádorská republika',
              common: 'Ekvádor',
            },
            cym: {
              official: 'Gweriniaeth Ecwador',
              common: 'Ecwador',
            },
            deu: {
              official: 'Republik Ecuador',
              common: 'Ecuador',
            },
            est: {
              official: 'Ecuadori Vabariik',
              common: 'Ecuador',
            },
            fin: {
              official: 'Ecuadorin tasavalta',
              common: 'Ecuador',
            },
            fra: {
              official: "République de l'Équateur",
              common: 'Équateur',
            },
            hrv: {
              official: 'Republika Ekvador',
              common: 'Ekvador',
            },
            hun: {
              official: 'Ecuadori Köztársaság',
              common: 'Ecuador',
            },
            ita: {
              official: "Repubblica dell'Ecuador",
              common: 'Ecuador',
            },
            jpn: {
              official: 'エクアドル共和国',
              common: 'エクアドル',
            },
            kor: {
              official: '에콰도르 공화국',
              common: '에콰도르',
            },
            nld: {
              official: 'Republiek Ecuador',
              common: 'Ecuador',
            },
            per: {
              official: 'جمهوری اکوادور',
              common: 'اکوادور',
            },
            pol: {
              official: 'Ekwador',
              common: 'Ekwador',
            },
            por: {
              official: 'República do Equador',
              common: 'Equador',
            },
            rus: {
              official: 'Республика Эквадор',
              common: 'Эквадор',
            },
            slk: {
              official: 'Ekvádorská republika',
              common: 'Ekvádor',
            },
            spa: {
              official: 'República del Ecuador',
              common: 'Ecuador',
            },
            swe: {
              official: 'Republiken Ecuador',
              common: 'Ecuador',
            },
            urd: {
              official: 'جمہوریہ ایکوڈور',
              common: 'ایکواڈور',
            },
            zho: {
              official: '厄瓜多尔共和国',
              common: '厄瓜多尔',
            },
          },
          latlng: [
            -2,
            -77.5,
          ],
          landlocked: false,
          borders: [
            'COL',
            'PER',
          ],
          area: 276841,
          demonyms: {
            eng: {
              f: 'Ecuadorean',
              m: 'Ecuadorean',
            },
            fra: {
              f: 'Équatorienne',
              m: 'Équatorien',
            },
          },
          flag: '🇪🇨',
          maps: {
            googleMaps: 'https://goo.gl/maps/TbX8hUW4gcbRPZiK7',
            openStreetMaps: 'https://www.openstreetmap.org/relation/108089',
          },
          population: 17643060,
          gini: {
            2019: 45.7,
          },
          fifa: 'ECU',
          car: {
            signs: [
              'EC',
            ],
            side: 'right',
          },
          timezones: [
            'UTC-06:00',
            'UTC-05:00',
          ],
          continents: [
            'South America',
          ],
          flags: [
            'https://flagcdn.com/ec.svg',
            'https://flagcdn.com/w320/ec.png',
          ],
        },
        loading: 'idle',
        error: null,
      },
    };

    const store = mockStore(mockState);
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <CountryDetails />
        </MemoryRouter>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
