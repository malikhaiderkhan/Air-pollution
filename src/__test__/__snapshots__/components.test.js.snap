// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing Country details should render correctly 1`] = `
<DocumentFragment>
  <div
    class="country-details"
  >
    <img
      alt="Ecuador"
      class="country-flag"
      src="https://flagcdn.com/w320/ec.png"
    />
    <p
      class="country-detail"
    >
      <b>
        Name
      </b>
      : Ecuador
    </p>
    <p
      class="country-detail"
    >
      <b>
        Population
      </b>
      : 17643060
    </p>
    <p
      class="country-detail"
    >
      <b>
        Capital
      </b>
      : Quito
    </p>
    <p
      class="country-detail"
    >
      <b>
        Region
      </b>
      : Americas
    </p>
    <p
      class="country-detail"
    >
      <b>
        Subregion
      </b>
      : South America
    </p>
    <p
      class="country-detail"
    >
      <b>
        Demonym
      </b>
      : Ecuadorean
    </p>
    <p
      class="country-detail"
    >
      <b>
        Landlocked
      </b>
      : No
    </p>
    <p
      class="country-detail"
    >
      <b>
        Area
      </b>
      : 276841
    </p>
    <p
      class="country-detail"
    >
      <b>
        Continent
      </b>
      : South America
    </p>
  </div>
</DocumentFragment>
`;

exports[`Component testing Country list should render correctly 1`] = `
<DocumentFragment>
  <div
    class="country-list-container"
  >
    <h2>
      Country Population
    </h2>
    <div
      class="country-search"
    >
      <input
        placeholder="Search country..."
        type="text"
        value=""
      />
    </div>
    <div
      class="country-grid"
    >
      <div
        class="country-box"
      >
        <a
          class="country-button"
          href="/country/ECU"
        >
          <img
            alt="Ecuador"
            src="https://flagcdn.com/w320/ec.png"
          />
          <h3>
            Ecuador
          </h3>
          <p>
            Population:17643060
          </p>
        </a>
      </div>
      <div
        class="country-box"
      >
        <a
          class="country-button"
          href="/country/FRA"
        >
          <img
            alt="France"
            src="https://flagcdn.com/w320/fr.png"
          />
          <h3>
            France
          </h3>
          <p>
            Population:67391582
          </p>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
